src/parser/jobs/war/modules/PrimalChaos.tsx

```diff
@@ -0,0 +1,117 @@
+import {Plural, Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {StatusKey} from 'data/STATUSES'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const DROPSY_STATUSES: StatusKey[] = [
+	'NASCENT_CHAOS',
+	'PRIMAL_REND_READY',
+]
+
+export class PrimalChaos extends Analyser {
+	static override handle = 'primalchaos'
+
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+
+	private droppedChaos: number = 0
+	private droppedRends: number = 0
+	private usedChaos: boolean = false
+	private usedRend: boolean = false
+
+	private chaosHook?: EventHook<Events['action']>
+	private primalHook?: EventHook<Events['action']>
+
+	override initialise() {
+		const dropsyStatuses = DROPSY_STATUSES.map(key => this.data.statuses[key].id)
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		this.addEventHook(playerFilter.type('statusApply').status(oneOf(dropsyStatuses)), this.onGain)
+		this.addEventHook(playerFilter.type('statusRemove').status(oneOf(dropsyStatuses)), this.onDrop)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onGain(event: Events['statusApply']) {
+		switch (event.status) {
+		case this.data.statuses.NASCENT_CHAOS.id:
```

Comment:
Personally, I would find separate functions for onGainNascentChaos and onGainPrimalRend rather than these switches, for readability reasons.  I don't know how much of a performance different it makes between either approach.

Reply:
It was to reduce hooks mostly. Maybe an ack question.